app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/btle/BtLEQueue.java

```diff
@@ -53,20 +57,27 @@
     private static final Logger LOG = LoggerFactory.getLogger(BtLEQueue.class);
 
     private final Object mGattMonitor = new Object();
+    private final Object mTransactionMonitor = new Object();
     private final GBDevice mGbDevice;
     private final BluetoothAdapter mBluetoothAdapter;
     private BluetoothGatt mBluetoothGatt;
+    private BluetoothGattServer mBluetoothGattServer;
+    private final Set<BluetoothGattService> mSupportedServerServices;
 
-    private final BlockingQueue<Transaction> mTransactions = new LinkedBlockingQueue<>();
+    private final Queue<Transaction> mTransactions = new ConcurrentLinkedQueue<>();
+    private final Queue<ServerTransaction> mServerTransactions = new ConcurrentLinkedQueue<>();
```

Comment:
Have you considered using a single queue for both client and server transactions? We could keep using the LinkedBlockingQueue, spare the mTransactionMonitor, the separate wait() and notify().

Reply:
Yes, that was my original plan. However, I did not succeed because, at a later stage, I couldn't distinguish between client and server operations anymore and ended up with a mess. I can give it a second try, though.

Reply:
OK, I took another approach now and I'm back to one queue and the old queue handling stuff.